src/Resistor/ResistorInput.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import { ResistorColorEntry as Color, Resistor } from 'puzzle-lib';
+import { DropdownButton, MenuItem } from 'react-bootstrap';
+import './ResistorInput.css';
+
+type Props = {
+};
+
+type State = {
+  value: String,
+  bands: (Color | null)[],
+  colorTable: Color[]
+};
+
+class ResistorInput extends React.Component<Props, State> {
+
+  private static readonly NULL_COLOR = new Color('None', '#DEC0DE', 0, undefined, undefined);

```

Comment:
nit: You can leave off the last two `undefined` parameters, that's the default anyway.